lib/screens/b_quiz/leaderBoard/leaderboard_repository.dart

```diff
@@ -0,0 +1,48 @@
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:ecellapp/models/leader_board.dart';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+
+import 'package:ecellapp/core/res/errors.dart';
+import 'package:ecellapp/core/res/strings.dart';
+import 'package:ecellapp/core/utils/injection.dart';
+import 'package:ecellapp/core/utils/logger.dart';
+
+
+@immutable
+abstract class LeaderRepository {
+  /// All subfunctions are final No arguments required returns json
+  Future<List<Data>> getAllleaders();
+}
+
+
+
+class APILeaderRepository extends LeaderRepository {
```

Comment:
Although you have included APIrepository, we also need the FakeAPI repo for any error simulation since currently we are not unit testing right now

Reply:
Ok sir